packages/react-ui/src/widgets/HistogramWidgetUI/HistogramWidgetUI.js

```diff
@@ -0,0 +1,352 @@
+      },
+      backgroundColor: theme.palette.other.tooltip,
+      confine: true,
+      position: 'top',
+      formatter(params) {
+        return tooltipFormatter(params, xAxisFormatter, yAxisFormatter);
+      }
+    }),
+    [theme, tooltip, tooltipFormatter, xAxisFormatter, yAxisFormatter]
+  );
+
+  // xAxis
+  const xAxisOptions = useMemo(
+    () => ({
+      min,
+      max,
+      interval: (max - min) / formattedData.length,
+      axisLine: {
+        show: false
+      },
+      splitLine: {
+        show: true,
+        lineStyle: {
+          color: theme.palette.charts.axisLine
+        }
+      },
+      axisTick: {
+        show: false
+      },
+      axisLabel: {
+        showMinLabel: true,
+        showMaxLabel: true,
+        ...theme.typography.charts,
+        hideOverlap: true,
+        padding: [theme.spacing(0.5), theme.spacing(0.5), 0, theme.spacing(0.5)],
+        formatter: (value) => {
+          const formattedValue = xAxisFormatter(value);
+          return value === min
+            ? formatMin(formattedValue)
+            : value === max
+            ? formatMax(formattedValue)
+            : formattedValue;
+        },
+        color: theme.palette.charts.maxLabel
+      }
+    }),
+    [min, max, formattedData.length, theme, xAxisFormatter]
+  );
+
+  // yAxis
+  const yAxisOptions = useMemo(
+    () => ({
+      type: 'value',
+      axisLine: {
+        show: false
+      },
+      splitLine: {
+        show: true,
+        lineStyle: {
+          color: theme.palette.charts.axisLine
+        }
+      },
+      axisTick: {
+        show: false
+      },
+      axisLabel: {
+        margin: 0,
+        verticalAlign: 'bottom',
+        padding: [0, 0, theme.typography.charts.fontSize, 0],
+        show: true,
+        showMaxLabel: true,
+        showMinLabel: false,
+        inside: true,
+        color: (value) => {
+          const maxValue = Math.max(...data.map((d) => d || Number.MIN_SAFE_INTEGER));
+          let col = 'transparent';
+          if (value >= maxValue) {
+            col = theme.palette.charts.maxLabel;
+          }
+
+          return col;
+        },
+        ...theme.typography.charts,
+        formatter: (v) => processFormatterRes(yAxisFormatter(v))
+      }
+    }),
+    [
+      theme.palette.charts.axisLine,
+      theme.palette.charts.maxLabel,
+      theme.typography.charts,
+      data,
+      yAxisFormatter
+    ]
+  );
+
+  // Series
+  const seriesOptions = useMemo(() => {
+    const dataWithColor = formattedData.map((item, idx) => {
+      const isDisabled = selectedBars.length && selectedBars.indexOf(idx) === -1;
+      const color = isDisabled
+        ? theme.palette.charts.disabled
+        : theme.palette.secondary.main;
+
+      return { value: item, itemStyle: { color } };
+    });
+
+    return {
+      type: 'custom',
+      cursor: 'pointer',
+      markArea: markAreaOptions,
+      renderItem: function (params, api) {
```

Comment:
Probably my lack of knowledge of some internals, but it's a bit hard to understand what value 0, 1 and 2 from api.value. I don't know if it exists at all, but it would be nice to have something like: `var [xxxxxx, xxxxxx, yValue] = api.values()  // or something similar` to have a good name per each one

Reply:
I copied this fn without knowing each details, but it's true, I'll improve this fn readability.